bridge/src/log_stream.rs

```diff
@@ -119,53 +126,31 @@ impl<T: Transport> Stream for LogStream<T> {
     fn poll(&mut self) -> Poll<Option<Self::Item>, Self::Error> {
         loop {
             let (next_state, value_to_yield) = match self.state {
-                State::AwaitInterval => {
-                    // wait until `interval` has passed
-                    let _ = try_stream!(self.poll_interval.poll().chain_err(|| format!(
-                        "LogStream (topic: #{:?}): polling interval failed",
-                        self.topic
-                    )));
-                    info!(
-                        "LogStream (topic: #{:?}): polling last block number",
-                        self.topic
+                State::AwaitBlockNumber => {
+                    let last_block = try_stream!(
+                        self.block_number_stream
+                            .poll()
+                            .chain_err(|| "LogStream: fetching of last confirmed block number failed")
                     );
-                    let future = web3::api::Eth::new(&self.transport).block_number();
-                    let next_state = State::AwaitBlockNumber(
-                        self.timer.timeout(future.from_err(), self.request_timeout),
+                    info!("LogStream: fetched confirmed block number {}", last_block);
+
+                    let from = self.last_checked_block + 1;
```

Comment:
I guess this kind of approach my not handle re-orgs well. I.e. you look for logs in say (4, 6) and then you will only look in (6, 7) even though 5,6 could actually re-organize, but `block_stream` does not return any new blocks for you.

I suppose it's fine though, cause we have the `confirmations` in the block stream, so it should only return blocks that are unlikely to re-organize.

Reply:
yes, the assumption is that we reorgs are < `confirmations`